### PR TITLE
Fix short list syntax detection.

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4356

### CODE REVIEW

File: tests/Fixer/ListNotation/ListSyntaxFixerTest.php [hunk 1]
```diff
@@ -88,6 +88,17 @@ public function provideToLongCases()
 ;#',
         ];
 
+        $cases[] = ['<?php
+
+class Test
+{
+    public function updateAttributeKey($key, $value)
+    {
+        $this->{camel_case($attributes)}[$key] = $value;
```
Comment (kubawerlos) [inline tests/Fixer/ListNotation/ListSyntaxFixerTest.php]: Is `camel_case` intentional? Got me confused seeing words "camel case" in snake case ;)

Comment (SpacePossum) [reply tests/Fixer/ListNotation/ListSyntaxFixerTest.php]: it is a copy paste of the reported sample that failed, I hadn't notice this until you pointed it out ;)

Comment (ikari7789) [reply tests/Fixer/ListNotation/ListSyntaxFixerTest.php]: Sorry, was a copy + paste from our Laravel app. `camel_case()` is a helper function. ^^;

File: src/Tokenizer/Transformer/SquareBraceTransformer.php [hunk 2]
```diff
@@ -44,6 +44,15 @@ public function getCustomTokens()
         ];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must run after CurlyBraceTransformer
```
Comment (SpacePossum) [inline src/Tokenizer/Transformer/SquareBraceTransformer.php]: both `BraceClassInstantiationTransformer` and `CurlyBraceTransformer` check for `CT::T_ARRAY_SQUARE_BRACE_CLOSE` which comes from this transformer, so there is a priority issue...

**Comment by SpacePossum** [comment]
I looked into the priority issue some more. The ideal situation would be:

```
CurlyBraceTransformer
	after SquareBraceTransformer because of:
		T_ARRAY_SQUARE_BRACE_CLOSE

SquareBraceTransformer
	after CurlyBraceTransformer because of:
		CT::T_DYNAMIC_PROP_BRACE_CLOSE
		CT::T_DYNAMIC_VAR_BRACE_CLOSE
		CT::T_ARRAY_INDEX_CURLY_BRACE_CLOSE
```

however, without splitting up the transformers this cannot be resolved.
The reason why CurlyBraceTransformer needs T_ARRAY_SQUARE_BRACE_CLOSE is for cases like:

```php
<?php $a = $b[0]{1};
```

Currently this is not an issue, because the transformer already/also accepts `]`.
The only other meaning `]` that could become an issue would be destructuring;

```php
<?php [$a]{1} = 1;
```

however these cases are always invalid in syntax and during execution.

**Pull request merged**